miden-lib/asm/miden/kernels/tx/tx.masm

```diff
@@ -55,6 +55,9 @@ const.ERR_NOTE_FUNGIBLE_MAX_AMOUNT_EXCEEDED=0x00020050
 # Non-fungible asset that already exists in the note cannot be added again
 const.ERR_NON_FUNGIBLE_ASSET_ALREADY_EXISTS=0x00020051
 
+# Note idx must be within [0, num_of_notes]
+const.ERR_INVALID_NOTE_IDX=0x00020052
```

Comment:
We should also add this error on the Rust side (i.e., [here](https://github.com/0xPolygonMiden/miden-base/blob/next/miden-tx/src/error.rs#L170)).